File: Zero para cancelar.c
#include <stdio.h>

int main() {
    int n; // Quantidade de números falados
    scanf("%d", &n);
    
    int pilha[n]; // Pilha para armazenar os números válidos
    int topo = 0; // Indica o índice do topo da pilha
    
    for (int i = 0; i < n; i++) {
        int x;
        scanf("%d", &x);
        
        if (x == 0 && topo > 0) {
            // Remove o último número válido (desempilhar)
            topo--;
        } else if (x != 0) {
            // Adiciona o número na pilha (empilhar)
            pilha[topo] = x;
            topo++;
        }
    }
    
    // Calcula a soma dos números na pilha
    int soma = 0;
    for (int i = 0; i < topo; i++) {
        soma += pilha[i];
    }
    
    printf("%d\n", soma); // Saída final
    return 0;
}

File: Zero para cancelar.py
n = int(input())  # Lê o número total de entradas
pilha = []  # Lista usada como pilha para armazenar os números válidos

# Lê os números um por um
for _ in range(n):
    x = int(input())
    if x == 0 and pilha:
        # Se o número é zero, remove o último número da pilha (desempilha)
        pilha.pop()
    elif x != 0:
        # Se o número é diferente de zero, adiciona na pilha (empilha)
        pilha.append(x)

# Calcula a soma dos números restantes na pilha
soma = sum(pilha)
print(soma)  # Exibe o resultado final
